overlord/snapstate/autorefresh.go

```diff
@@ -73,6 +75,19 @@ var (
 	}()
 )
 
+type statusCheckerData struct {
+	info    *snap.Info
```

Comment:
we never keep infos around for long this way, as many operations that are not visible to this code could move the revision of the snap around which would change the info. In general this is a problem of this code as is, is too unaware of what the rest of snapd is up to.

Reply:
I'll change it to only monitor the PIDs, then.

Reply:
We need to step back and all have a chat also with @mvo5 what we can achieve short term and medium term and how. The spec was in a state close to drive implementation.

Reply:
I removed the need for the Info structure. Now it just checks the CGroups for the corresponding InstanceName, so everything is static.